pkg/lock.go

```diff
@@ -0,0 +1,273 @@
+		"config",
+		"--host", configSVRDSN,
+		"--quiet",
+		"--eval", "db.BackupControl.findAndModify({query: { _id: 'BackupControlDocument' }, update: { $inc: { counter : 1 } }, new: true, upsert: true, writeConcern: { w: 'majority', wtimeout: 15000 }});",
+	}, mongoCreds...)
+
+	output, err := sh.Command(MongoCMD, args...).Output()
+	if err != nil {
+		klog.Errorf("Error while running findAndModify to setup configServer : %s ; output : %s \n", err.Error(), output)
+		return err
+	}
+
+	err = json.Unmarshal(output, &v)
+	if err != nil {
+		klog.Errorf("Unmarshal error while running findAndModify to setup configServer : %s \n", err.Error())
+		return err
+	}
+	val, ok := v["counter"].(float64)
+	if !ok || int(val) == 0 {
+		return fmt.Errorf("unable to modify BackupControlDocument. got response: %v", v)
+	}
+	val2 := float64(0)
+	timer := 0 // wait approximately 5 minutes.
+	for timer < 60 && (int(val2) == 0 || int(val) != int(val2)) {
+		timer++
+		// find backupDocument from secondary configServer
+		args = append([]interface{}{
+			"config",
+			"--host", secondaryHost,
+			"--quiet",
+			"--eval", "rs.secondaryOk(); db.BackupControl.find({ '_id' : 'BackupControlDocument' }).readConcern('majority');",
+		}, mongoCreds...)
+
+		if err := sh.Command(MongoCMD, args...).UnmarshalJSON(&v); err != nil {
+			return err
+		}
+
+		val2, ok = v["counter"].(float64)
+		if !ok {
+			return fmt.Errorf("unable to get BackupControlDocument. got response: %v", v)
+		}
+		if int(val) != int(val2) {
+			klog.V(5).Infof("BackupDocument counter in secondary %v is not same. Expected %v, but got %v. Full response: %v", secondaryHost, val, val2, v)
+			time.Sleep(time.Second * 5)
+		}
+	}
+	if timer >= 60 {
+		return fmt.Errorf("timeout while waiting for BackupDocument counter in secondary %v to be same as primary. Expected %v, but got %v. Full response: %v", secondaryHost, val, val2, v)
+	}
+
+	return nil
+}
+
+func lockSecondaryMember(mongohost string) error {
+	klog.Infoln("Attempting to lock secondary member", mongohost)
+
+	if mongohost == "" {
+		klog.Warningln("locking secondary member is skipped. secondary host is empty")
+		return nil
+	}
+
+	// lock file
+	v := make(map[string]interface{})
+	args := append([]interface{}{
+		"config",
+		"--host", mongohost,
+		"--quiet",
+		"--eval", "JSON.stringify(db.fsyncLock())",
+	}, mongoCreds...)
+
+	output, err := sh.Command(MongoCMD, args...).Output()
+	if err != nil {
+		klog.Errorf("Error while running fsyncLock on secondary : %s ; output : %s \n", err.Error(), output)
+		return err
+	}
+
+	err = json.Unmarshal(output, &v)
+	if err != nil {
+		klog.Errorf("Unmarshal error while running fsyncLock on secondary : %s \n", err.Error())
+		return err
+	}
+
+	if val, ok := v["ok"].(float64); !ok || int(val) != 1 {
+		return fmt.Errorf("unable to lock the secondary host. got response: %v", v)
+	}
+	klog.Infof("secondary %s locked.", mongohost)
+	return nil
+}
+
+func checkIfSecondaryLockedAndSync(mongohost string) error {
```

Comment:
We need to do it for all secondaries. Because, In provisioner operator , we will check for all , not a single secondary.